lib/datadog_backup/resources/class_methods.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module DatadogBackup
+  class Resources
+    module ClassMethods
+      def new_resource(id: nil, body: nil)
+        raise ArgumentError, 'id and body cannot both be nil' if id.nil? && body.nil?
+
+        new(
+          id: id,
+          body: body,
+          api_version: @api_version,
+          api_resource_name: @api_resource_name,
+          id_keyname: @id_keyname,
+          banlist: @banlist,
+          api_service: @api_service
+        )
+      end
+
+      def all
+        @all ||= get_all.map do |resource|
+          new_resource(id: resource.fetch(@id_keyname), body: resource)
+        end
+        LOGGER.info "Found #{@all.length} #{@api_resource_name}s in Datadog"
```

Comment:
This logging will happen even if the `get_all` response has already been cached in the `@all` instance variable.  Is that intended?